include/kframework/foundry.md

```diff
@@ -212,21 +212,21 @@ module FOUNDRY-CHEAT-CODES
 
 First we have some helpers in K which can:
 
--   Inject a given boolean condition into's this execution's path condition, and
--   Check that a given boolean condition holds (recording failure if not).
-
+-   Inject a given boolean condition into's this execution's path condition
+-   Set the `FoundryCheat . Failed` location to `True`.
 ```k
     syntax KItem ::= #assume ( Bool ) [klabel(foundry_assume), symbol]
-                   | #assert ( Bool ) [klabel(foundry_assert), symbol]
  // ------------------------------------------------------------------
     rule <k> #assume(B) => . ... </k> ensures B
 
-    rule <k> #assert(false) => . ... </k>
-         <account>
-           <acctID> #address(FoundryCheat) </acctID>
-           <storage> STORAGE => STORAGE [ #loc(FoundryCheat . Failed) <- 1 ] </storage>
-           ...
-         </account>
+     syntax KItem ::= "#markAsFailed" [klabel(foundry_markAsFailed)]
+  // ---------------------------------------------------------------
+     rule <k> #markAsFailed => . ... </k>
+          <account>
+             <acctID> #address(FoundryCheat) </acctID>
+             <storage> STORAGE => STORAGE [ #loc(FoundryCheat . Failed) <- 1 ] </storage>
+             ...
+           </account>
```

Comment:
Is this how Foundry works? It marks as failed for when an expectRevert fails? Or does it have a separate way of storing that?

Also, if we are marking as failed like this, do we need the `ISREVERTEXPECTED` to be passed into `foundry_success` anymore?

Reply:
Well, `expectRevert` looks at the next call made after the invocation of the cheat code. If I'm not setting the storage location, then any upcoming revert call could clear the `ISREVERTEXPECTED`, at least that was my understanding.

>Also, if we are marking as failed like this, do we need the ISREVERTEXPECTED to be passed into foundry_success anymore?

The `ISREVERTEXPECTED` is still passed to the `foundry_success` predicate in the case in which the test ends with a pending `expectRevert` (i.e. the cheat code is invoked but no further call is made).